katsdptelstate/test/test_telescope_state.py

```diff
@@ -504,6 +504,15 @@ def test_undecodable_bytes_in_key(self):
         self.assertEqual(self.ts.get(key_b), 'hello')
         self.assertEqual(self.ts.get(key_b[1:]), None)
 
+    def test_tab_completion(self):
+        self.ts.clear()
+        ns = self.ts.view('ns')
+        ns['c'] = 'value'
+        ns['b'] = 'value'
+        ns['a'] = 'value'
+        keys = ns._ipython_key_completions_()
+        self.assertEqual(sorted(keys), ['a', 'b', 'c', 'ns_a', 'ns_b', 'ns_c'])
```

Comment:
Also add a key in the top-level namespace to ensure that non-leaf keys are handled correctly.

Reply:
Good point.